warehouse/query-core/src/main/java/datawave/query/language/parser/jexl/JexlNodeSet.java

```diff
@@ -22,25 +18,39 @@
 /**
  * Utility class that implements the {@link Set} interface for use with a collection of Jexl nodes.
  *
- * Element uniqueness is computed using {@link JexlASTHelper#nodeToKey(JexlNode)} for keys.
+ * Nodes are stored in an underlying map of String keys to JexlNode values. If the underlying nodes are changed then the set is considered invalid.
  *
- * If a Jexl node and it's {@link ASTDelayedPredicate} equivalent are both added to the set, the set will accept the delayed predicate.
+ * The node keys are built using {@link JexlASTHelper#nodeToKey(JexlNode)}.
  *
- * The JexlNodeSet is considered invalid if any of the underlying Jexl nodes change.
+ * If the {@link #useSourceNodeForKeys} flag is set then {@link datawave.query.jexl.nodes.QueryPropertyMarker} nodes will generate a node key from the unwrapped
+ * source node.
+ *
+ * For example, (ASTDelayedPredicate &amp;&amp; FOO == 'bar') will be unwrapped to FOO == 'bar', generating the node key "FOO == 'bar'".
  */
 public class JexlNodeSet implements Set<JexlNode> {
     
+    // Determines if we unwrap QueryPropertyMarker nodes and use their source node to determine node uniqueness.
+    private boolean useSourceNodeForKeys = false;
+    
     // Internal map of node keys to nodes;
     private final Map<String,JexlNode> nodeMap;
     
-    private static final Logger log = Logger.getLogger(JexlNodeSet.class);
-    
     public JexlNodeSet() {
+        this(false);
```

Comment:
I hate to ask this, but why not make the default 'true' since that seems to be how this is used everywhere?

Reply:
In the index lookup code node equality is synonymous with unwrapping QueryPropertyMarker nodes. If you feel this should be the default behavior everywhere I'm comfortable making that change.

Someone using this class for the first time might reasonably assume that an EQNode and the wrapped version of that EQNode are separate and distinct nodes.